util/dvsim/doc/testplanner.md

```diff
@@ -0,0 +1,285 @@
+
+-------
+  // Imported testplan:
+  {
+    name: csr
+    ...
+    tests: ["{name}{intf}_csr_hw_reset_{foo}"]
+  }
+```
+
+This will resolve to the following 6 tests:
+
+```
+["chip_csr_hw_reset_x", "chip_csr_hw_reset_y", "chip_csr_hw_reset_z",
+ "chip_jtag_csr_hw_reset_x", "chip_jtag_csr_hw_reset_y", "chip_jtag_csr_hw_reset_z"]
+```
+
+### Example sources
+
+The following examples provided within `util/dvsim/examples/testplanner` can be
+used as a starting point.
+* **`foo_testplan.hjson`**: DUT testplan
+* **`common_testplan.hjson`**: shared testplan imported within the DUT testplan
+* **`foo_dv_doc.md`**: DUT testplan imported within the DV document doc in
+  Markdown
+
+In addition, see the [UART DV document]({{< relref "hw/ip/uart/doc/dv" >}}) for
+a 'production' example of inline expansion of an imported testplan as a table
+within the DV document.
+The [UART testplan](https://github.com/lowRISC/opentitan/blob/master/hw/ip/uart/data/uart_testplan.hjson)
+imports the shared testplans located at `hw/dv/tools/dvsim/testplans` area.
+
+### Limitations
+
+The following limitations currently hold:
+* Only the DUT testplan can import shared testplans; the imported
+  testplans cannot further import more testplans
+* All planned test names parsed from the DUT testplan and all of
+  its imported tetsplans need to be unique
+
+## Usage examples
+
+### Standalone tool invocations
+
+Generate the testplan table in HTML to stdout:
+```console
+$ cd $REPO_TOP
+$ ./util/dvsim/testplanner.py \
+    util/dvsim/examples/testplanner/foo_testplan.hjson
+```
+
+Generate the testplan table in HTML to a file:
+```console
+$ ./util/dvsim/testplanner.py \
+    util/dvsim/examples/testplanner/foo_testplan.hjson \
+    -o /tmp/foo_testplan_table.html
+```
+
+Generate simulation result tables in HTML to stdout:
+```console
+$ ./util/dvsim/testplanner.py \
+    util/dvsim/examples/testplanner/foo_testplan.hjson \
+    -s util/dvsim/examples/testplanner/foo_sim_results.hjson
+```
+
+Generate simulation result tables in HTML to a file:
+```console
+$ ./util/dvsim/testplanner.py \
+    util/dvsim/examples/testplanner/foo_testplan.hjson \
+    -s util/dvsim/examples/testplanner/foo_sim_results.hjson \
+    -o /tmp/foo_sim_results.html
+```
+
+### APIs for external tools
+The `util/build_docs.py` script invokes the testplanner utility functions
+directly to parse the Hjson testplan and insert a HTML table within the DV
+plan document. This is done by invoking:
+
+```console
+$ ./util/build_docs.py --preview
+```
+The output for each testplan will be saved into `build/docs-generated`.
+For example the path to the GPIO IP testplan is
+`build/docs-generated/hw/ip/gpio/data/gpio_testplan.hjson.testplan`.
+
+See following snippet of code for the APIs in use:
+```python
+from dvsim.Testplan import Testplan
+
+  # hjson_testplan_path: a string pointing to the path to Hjson testplan
+  testplan = Testplan(hjson_testplan_path)
+  text = testplan.get_testplan_table("html")
+```
```

Comment:
I'd suggest just pointing at some code that uses the APIs. Otherwise, you have to remember to update this document if the APIs change and nothing will break if you don't. In my experience, I **always** forget, so think it's better not to need to worry about it :-)